tools/ports/harfbuzz.py

```diff
@@ -0,0 +1,38 @@
+import os, shutil, logging, subprocess
+
+TAG = '1.7.5'
+
+def get(ports, settings, shared):
+  if settings.USE_HARFBUZZ == 1:
+    ports.fetch_project('harfbuzz', 'https://github.com/harfbuzz/harfbuzz/archive/' + TAG + '.zip', 'harfbuzz-' + TAG)
+    def create():
+      logging.info('building port: harfbuzz')
+
+      source_path = os.path.join(ports.get_dir(), 'harfbuzz', 'harfbuzz-' + TAG)
+      dest_path = os.path.join(shared.Cache.get_path('ports-builds'), 'harfbuzz')
+
+      shutil.rmtree(dest_path, ignore_errors=True)
+      shutil.copytree(source_path, dest_path)
+
+      cmake_build_type = 'Release'
+      subprocess.check_call(['cmake', '-Bbuild', '-H.', '-DCMAKE_BUILD_TYPE=' + cmake_build_type])
+      subprocess.check_call(['make', '-Cbuild'])
```

Comment:
I am issuing `cmake` and `make` command by my own as cmake of harfbuzz has limitation on in-tree make file generation (which I did that while adding the cmake support to not conflict with autotools based make files). I know it is not clean at all, what we can do?

Reply:
Maybe I'm missing something, but why is this bad? Seems like it creates a build dir and builds in there, and all this is done under `ports-builds/harfbuzz` which is the proper place, all of which is ok I think.

Reply:
Great!